builtin/game/item.lua

```diff
@@ -38,6 +38,10 @@ function core.get_pointed_thing_position(pointed_thing, above)
 end
 
 function core.dir_to_facedir(dir, is6d)
+	-- nil check for dir (for use with place_node)
+	if not dir then
+		return 0
```

Comment:
are you sure it's correct to return a wrong value ? a failure seems more lgtm as it's a caller problem

Reply:
So long as it stops the error either way is fine, this was just a quick fix that worked.

Reply:
@paramat can you look at this issue to fix the caller problem, instead of doing a hack in this function ?

Reply:
@tenplus1 to do a C++ example you check is like you did

```
Object *obj = null;
obj->set_attr("blah");
```
and:

```
int Object::set_attr(string str) {
  if (this == null) {
    return 0;
  }
  m_str = str;
  return 1;
}
```

Reply:
I'm not comfortable writing in c++ and for the sake of a quick check this pull fixes the issue until someone can do it properly in the engine.